pennylane/_qubit_device.py

```diff
@@ -359,8 +302,48 @@ def marginal_prob(self, prob, wires=None):
         Returns:
             list[float]: List of the resulting marginal probabilities.
         """
-        wires = wires or range(self.num_wires)
+        wires = list(wires or range(self.num_wires))
         wires = np.hstack(wires)
         inactive_wires = list(set(range(self.num_wires)) - set(wires))
         prob = prob.reshape([2] * self.num_wires)
         return np.apply_over_axes(np.sum, prob, inactive_wires).flatten()
+
+    def expval(self, observable):
+        wires = observable.wires
+
+        if self.analytic:
+            # exact expectation value
+            eigvals = observable.eigvals
+            prob = self.probability(wires=wires)
+            return (eigvals @ prob).real
+
+        # estimate the ev
+        return np.mean(self.sample(observable))
+
+    def var(self, observable):
+        wires = observable.wires
+
+        if self.analytic:
+            # exact variance value
+            eigvals = observable.eigvals
+            prob = self.probability(wires=wires)
+            return (eigvals ** 2) @ prob - (eigvals @ prob).real ** 2
+
+        # estimate the variance
+        return np.var(self.sample(observable))
+
+    def sample(self, observable):
+        wires = observable.wires
+        name = observable.name
```

Comment:
Really nice to see that the ``expval``, ``var`` and ``sample`` methods don't have the conditional ``generate_samples`` kind of logic in there anymore! Looks way neater :)